common/stream.h

```diff
@@ -44,6 +45,19 @@ class IStream : public Object
         return writev(iov, iovcnt);
     }
 
+    struct ReadAll {
+        struct FreeDeleter {
+            void operator()(void* ptr) {
+                ::free(ptr);
+            }
+        };
+        std::unique_ptr<void, FreeDeleter> ptr;
+        ssize_t size;   // < 0 if error occured; |size| is always the # of bytes read
+    };
+
+    // read until EOF
+    ReadAll readall(size_t min_buf = 1024, size_t max_buf = 1024 * 1024 * 1024);
```

Comment:
is the param 'min_buf' necessary?

Reply:
in case that you want it start from, say, 1MB, to avoid some unnecessary realloc()